File: Bailaconmigo/src/main/java/org/example/bailaconmigo/DTOs/EventGeneralReportDto.java
package org.example.bailaconmigo.DTOs;

import lombok.Data;
import org.example.bailaconmigo.Entities.Enum.DanceStyle;
import org.example.bailaconmigo.Entities.Enum.EventType;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.Map;

@Data
public class EventGeneralReportDto {
    private LocalDateTime generatedAt;

    // Estadísticas generales de eventos
    private int totalEvents;
    private int activeEvents;
    private int cancelledEvents;

    // Estadísticas de inscripciones
    private int totalRegistrations;
    private int confirmedRegistrations;
    private int pendingRegistrations;
    private int cancelledRegistrations;

    // Métricas financieras
    private BigDecimal totalRevenue;
    private double averageRegistrationsPerEvent;

    // Distribuciones
    private Map<EventType, Long> eventsByType;
    private Map<DanceStyle, Long> danceStylePopularity;
}

File: frontend/src/app/models/event-general-report-dto.ts
import { EventType } from "./event-type";

export interface EventGeneralReportDto {
  generatedAt: string;
  totalEvents: number;
  activeEvents: number;
  cancelledEvents: number;
  totalRegistrations: number;
  confirmedRegistrations: number;
  pendingRegistrations: number;
  cancelledRegistrations: number;
  totalRevenue: number;
  averageRegistrationsPerEvent: number;
  eventsByType: Record<EventType, number>;
  danceStylePopularity: Record<string, number>;
}
